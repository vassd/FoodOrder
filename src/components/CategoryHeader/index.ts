export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
