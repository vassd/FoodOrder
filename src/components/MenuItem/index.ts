export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
