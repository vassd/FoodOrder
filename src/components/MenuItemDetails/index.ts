export { MenuItemDetails } from './MenuItemDetails'
export type { Topping, Crust } from './MenuItemDetails'
