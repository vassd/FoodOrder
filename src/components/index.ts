export { MenuHeader } from './MenuHeader'
export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
export { MenuItems } from './MenuItems'
export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
