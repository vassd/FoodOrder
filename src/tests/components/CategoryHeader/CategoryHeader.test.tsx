import React from 'react'
import '@testing-library/jest-dom'
import { describe, expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import { CategoryHeader } from 'components'

describe('CategoryHeader test', () => {
  test('should render correctly', () => {
    render(<CategoryHeader id={0} name={'pizza'} hasCrust hasToppings />)

    expect(screen.getByText('pizza')).toBeInTheDocument()
  })

  test('should match snapshot', () => {
    const { container } = render(
      <CategoryHeader id={0} name={'pizza'} hasCrust hasToppings />
    )

    expect(container).toMatchSnapshot()
  })
})
