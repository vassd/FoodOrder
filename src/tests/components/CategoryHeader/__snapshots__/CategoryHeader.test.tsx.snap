// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryHeader test > should match snapshot 1`] = `
<div>
  <div
    class="_category-header_d36887"
  >
    pizza
  </div>
</div>
`;
